### PR TITLE
test(core): Move unit tests closer to testable components (no-changelog)

### PR DESCRIPTION
**Pull request by tomi**
## Summary

Unit tests have been residing in `test/unit/` directory. We want to them to be closer to the source files. This moves all the unit tests to `__tests__` directories next to the components that are being tested.

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/cli/test/integration/PermissionChecker.test.ts [hunk 1]
```diff
@@ -25,6 +24,36 @@ import { ProjectRepository } from '@/databases/repositories/project.repository';
 
 const ownershipService = mockInstance(OwnershipService);
 
+function mockNodeTypesData(
```
Comment (tomi) [inline packages/cli/test/integration/PermissionChecker.test.ts:27]: Moved from `test/unit/Helpers.ts`

File: packages/cli/test/shared/flushPromises.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+/**
+ * Ensure all pending promises settle. The promise's `resolve` is placed in
+ * the macrotask queue and so called at the next iteration of the event loop
+ * after all promises in the microtask queue have settled first.
+ */
+export const flushPromises = async () => await new Promise(setImmediate);
```
Comment (tomi) [inline packages/cli/test/shared/flushPromises.ts:6]: Moved from `test/unit/Helpers.ts`

**Review by netroy — APPROVED** [review]
Thanks 🎉

**Comment by janober** [comment]
Got released with [`n8n@1.54.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.54.0)

**Pull request merged**